### PR TITLE
Feature/registation and authorization

### PR DESCRIPTION
**Pull request by AndriiSameliuk**
Implement services for working with sets of data from the UI part of a web application.

### CODE REVIEW

**Comment by AndriiSameliuk** [comment]
@BohdanDudkaTD, PTAL.

File: src/main/java/io/javaclasses/hookahpalace/service/AuthorizationService.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Makes queries to a database for getting necessary user or information about him.
+ */
+public class AuthorizationService {
+
+    /**
+     * Makes a query to a database for getting necessary user.
+     *
+     * @param credential id, login and password of user.
```
Comment (BohdanDudkaTD) [inline src/main/java/io/javaclasses/hookahpalace/service/AuthorizationService.java]: There is no `id` field in the Credentials.

File: src/test/java/io/javaclasses/hookahpalace/service/AuthorizationTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource("addAuthorizationTokens")
+    void testGettingInformationAboutUser(AuthorizationToken authorizationToken,
+                                         UserInfo expectedUserInfo,
+                                         String assertMessage) throws AuthorizationException {
+        Optional<UserInfo> actualUserInfo =
+                authorizationService.getUserInfo(authorizationToken);
+        actualUserInfo.ifPresent(userInfo ->
```
Comment (BohdanDudkaTD) [inline src/test/java/io/javaclasses/hookahpalace/service/AuthorizationTest.java]: What if optional is empty?

File: src/main/java/io/javaclasses/hookahpalace/AuthorizationException.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package io.javaclasses.hookahpalace;
+
+/**
+ * This is an exception which can be throw when input data invalid.
```
Comment (BohdanDudkaTD) [inline src/main/java/io/javaclasses/hookahpalace/AuthorizationException.java]: can be throwN when input data IS invalid.

File: src/main/java/io/javaclasses/hookahpalace/RegistrationException.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package io.javaclasses.hookahpalace;
+
+/**
+ * This is an exception which can be throw when input data invalid.
```
Comment (BohdanDudkaTD) [inline src/main/java/io/javaclasses/hookahpalace/RegistrationException.java]: can be throwN when input data IS invalid.

File: src/main/java/io/javaclasses/hookahpalace/ValidationException.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package io.javaclasses.hookahpalace;
+
+/**
+ * This is an exception which can be throw when input data invalid.
```
Comment (BohdanDudkaTD) [inline src/main/java/io/javaclasses/hookahpalace/ValidationException.java]: can be throwN when input data IS invalid.

**Pull request merged**